Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -534,54 +535,134 @@ define([
         };
     };
 
-    Cesium3DTileBatchTable.prototype.getFragmentShaderCallback = function() {
-        if (this.featuresLength === 0) {
-            return;
+    function replaceDiffuse(source, colorBlendMode, diffuseUniformName) {
+        source = ShaderSource.replaceMain(source, 'tile_main');
+        // If the color mode is highlight and the glTF does not specify the _3DTILESDIFFUSE semantic, return a basic highlight shader.
+        // Otherwise if _3DTILESDIFFUSE is defined prefer the shader below that can switch the color mode at runtime.
+        if ((colorBlendMode === Cesium3DTileColorBlendMode.HIGHLIGHT) && !defined(diffuseUniformName)) {
+            return source +
+                   'void tile_highlightDiffuse(vec4 tile_featureColor) \n' +
+                   '{ \n' +
+                   '    tile_main(); \n' +
+                   '    gl_FragColor *= tile_featureColor; \n' +
+                   '} \n';
         }
 
-        return function(source, handleTranslucent) {
-            //TODO: generate entire shader at runtime?
-            //var diffuse = 'diffuse = u_diffuse;';
-            //var diffuseTexture = 'diffuse = texture2D(u_diffuse, v_texcoord0);';
-            //if (ContextLimits.maximumVertexTextureImageUnits > 0) {
-            //    source = 'varying vec3 tile_featureColor; \n' + source;
-            //    source = source.replace(diffuse, 'diffuse.rgb = tile_featureColor;');
-            //    source = source.replace(diffuseTexture, 'diffuse.rgb = texture2D(u_diffuse, v_texcoord0).rgb * tile_featureColor;');
-            //} else {
-            //    source =
-            //        'uniform sampler2D tile_batchTexture; \n' +
-            //        'varying vec2 tile_featureSt; \n' +
-            //        source;
-            //
-            //    var readColor =
-            //        'vec4 featureProperties = texture2D(tile_batchTexture, tile_featureSt); \n' +
-            //        'if (featureProperties.a == 0.0) { \n' +
-            //        '    discard; \n' +
-            //        '}';
-            //
-            //    source = source.replace(diffuse, readColor + 'diffuse.rgb = featureProperties.rgb;');
-            //    source = source.replace(diffuseTexture, readColor + 'diffuse.rgb = texture2D(u_diffuse, v_texcoord0).rgb * featureProperties.rgb;');
-            //}
-            //
-            //return source;
-
-            // TODO: support both "replace" and "highlight" color?  Highlight is below, replace is commented out above
-            var renamedSource = ShaderSource.replaceMain(source, 'tile_main');
-            var newMain;
+        //>>includeStart('debug', pragmas.debug);
+        if (((colorBlendMode === Cesium3DTileColorBlendMode.REPLACE) || (colorBlendMode === Cesium3DTileColorBlendMode.MIX)) && !defined(diffuseUniformName)) {
+            throw new DeveloperError('The glTF diffuse parameter must have a _3DTILESDIFFUSE semantic to support the replace or mix color blend mode');
```

Comment:
Perhaps this could be a warning or "info" option in the validator.  If you agree, please update the roadmap.